test/unit/alarm.spec.js

```diff
@@ -0,0 +1,19 @@
+describe('Alarm', function() {
+
+  beforeEach(module('smartAlarm'));
+
+  beforeEach(inject(function($controller, $q){
+    deferredLogin = $q.defer();
+    testController = $controller('ExampleController');
+  }));
+
+  describe('add', function() {
+
+    it('calls on the NotificationScheduler factory', function () {
```

Comment:
Are we able to test this functionality at all? If not, we should eliminate the test?